src/gretel_trainer/relational/json.py

```diff
@@ -314,30 +314,48 @@ def _denormalize_json(
     table_name_mappings: dict[str, str],
     original_table_name: str,
 ) -> pd.DataFrame:
-    table_names = list(table_name_mappings.values())
+    # Throughout this function, "provenance name" refers to the json-breadcrumb-style name (e.g. foo^bar>baz)
```

Comment:
should this be a function docstring, or is that not appropriate since it also talks about how things are used below?

Reply:
(Disclaimer: I'm not as consistent about comments and/or docstrings as I ought to be.)

This feels more like a note to developers who need to grok the implementation for debugging/modifying/refactoring, as opposed to a note to clients of the function (who won't know or need to care about "provenance" vs "node"), so to me this makes more sense as a comment than a docstring.